File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p02683/p02683_num9_parsed.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    int N, M, X;
    cin >> N >> M >> X;

    vector<int> C(N);
    vector<vector<int>> A(N, vector<int>(M));

    for (int i = 0; i < N; i++) {
        cin >> C[i];
        for (int j = 0; j < M; j++) {
            cin >> A[i][j];
        }
    }

    int minCost = -1;

    for (int mask = 0; mask < (1 << N); mask++) {
        vector<int> skills(M, 0);
        int cost = 0;
        for (int i = 0; i < N; i++) {
            if (mask & (1 << i)) {
                for (int j = 0; j < M; j++) {
                    skills[j] += A[i][j];
                }
                cost += C[i];
            }
        }

        bool achieved = true;
        for (int j = 0; j < M; j++) {
            if (skills[j] < X) {
                achieved = false;
                break;
            }
        }

        if (achieved && (minCost == -1 || cost < minCost)) {
            minCost = cost;
        }
    }

    cout << minCost << endl;

    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p02683/p02683_num9_parsed.py
# Python program to solve the problem

# Function to check if objective is achievable
def check_objective(N, M, X, C, A):
    
    min_cost = float('inf')
    
    # Loop through all possible combinations of books
    for i in range(1, 2**N):
        cost = 0
        skills = [0] * M
        
        # Calculate the cost and skills for current combination
        for j in range(N):
            if i & (1 << j):
                cost += C[j]
                for k in range(M):
                    skills[k] += A[j][k]
        
        # Check if current combination satisfies the objective
        if all(skill >= X for skill in skills):
            min_cost = min(min_cost, cost)
    
    # Return the minimum cost if objective is achievable, else -1
    return min_cost if min_cost != float('inf') else -1

# Read the input values
N, M, X = map(int, input().split())
C = []
A = []
for _ in range(N):
    c, *a = map(int, input().split())
    C.append(c)
    A.append(a)

# Check if objective is achievable and print the result
result = check_objective(N, M, X, C, A)
print(result)